### ISSUE TITLE
bug: ShapeTree.title calls ph_idx  on non-placeholder shapes

### ISSUE THREAD

**Issue by joereddington** [scanny/python-pptx#153]
Hi, I'm trying to read the title from this file: https://www.dropbox.com/s/2q3m0hpzrxum8mj/CommuniKate20-es.pptx?dl=0 

with this code: 
```
#!/usr/bin/python
from pptx import Presentation

prs = Presentation("testSuite/test1/CommuniKate20-es.pptx")


for slide in prs.slides:
    title_placeholder = slide.shapes.title
    print "New Slide!"

```

but I get this ValueException immediately... :s 
'''
Traceback (most recent call last):
  File "./reduce.py", line 8, in <module>
    title_placeholder = slide.shapes.title
  File "/Library/Python/2.7/site-packages/python_pptx-0.5.7-py2.7.egg/pptx/shapes/shapetree.py", line 226, in title
    if elm.ph_idx == 0:
  File "/Library/Python/2.7/site-packages/python_pptx-0.5.7-py2.7.egg/pptx/oxml/shapes/shared.py", line 77, in ph_idx
    raise ValueError("not a placeholder shape")
ValueError: not a placeholder shape

'''

Bit confused, as the code was copied from https://www.dropbox.com/s/xuvhyzo26gfqtun/Screenshot%202015-03-21%2014.28.08.png?dl=0  - have I just muddled myself up or does this look like a bug? (either in the code or the docs...) 

Awesome tool by the way - saved me hours and hours so far... just finishing this script would be amazing... 

J

**Comment by scanny** [comment]
Hi Joe, yes, this actually does look like a bug. Line 225 in pptx/shapes/shapetree.py should be:
```python
for elm in self._spTree.iter_ph_elms():
```
instead of:
```python
for elm in self._spTree.iter_shape_elms():
```

You can work around this by adding your own title placeholder function like this:
```python
def slide_title_placeholder(slide):
    for shape in slide.shapes:
        if not shape.is_placeholder:
            continue
        if shape.placeholder_format.idx == 0:
            return shape
    return None
```

Keep in mind this function can return `None` because not all slides necessarily have a title placeholder.

I'll leave this issue open as a bug report to get this fixed up.

**Comment by joereddington** [comment]
Workaround worked perfectly :)

**Comment by scanny** [comment]
Fixed in v0.5.8.
